### PR TITLE
[None][fix] Use try/except fallback for Pydantic ValidatorIterator in chat message parsing

### PR DESCRIPTION
**Pull request by Wanli-Jiang**
# Why?
* tool_calls message in tau2bench_telecom is Pydantic v2 dtype, which will be failed when parsing messages.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved robustness of tool call handling in chat message parsing to prevent data mutation.
  * Added fallback mechanism for chat message parsing with graceful error recovery when validation issues occur.

* **Tests**
  * Comprehensive unit tests added for tool call parsing, including edge cases with iterators, extra fields, and error scenarios.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/serve/chat_utils.py [hunk 1]
```diff
@@ -244,8 +244,16 @@ def _parse_assistant_message_content(message: Dict[str, Any]) -> Dict[str, Any]:
...
     if tool_calls is not None:
+        # Materialize Pydantic v2 ValidatorIterator (single-use) to a list.
+        if not isinstance(tool_calls, list):
+            tool_calls = list(tool_calls)
+
         result["tool_calls"] = []
         for item in tool_calls:
+            item = dict(item)
+            if "function" in item:
+                item["function"] = dict(item["function"])
```
Comment (JunyiXu-nv) [inline tensorrt_llm/serve/chat_utils.py]: Could we add a comment here like: 
```
# Bypass pydantic check to WAR `tau2-bench-telecom` ill-format tool_call
```
As a note.

**Review by JunyiXu-nv — APPROVED** [review]

File: tensorrt_llm/serve/chat_utils.py (hunk 1 above)
Comment (Wanli-Jiang) [reply tensorrt_llm/serve/chat_utils.py]: updated

**Review by syuoni — APPROVED** [review]
